dipy/tracking/mesh.py

```diff
@@ -0,0 +1,176 @@
+
+import numpy as np
+
+
+def random_coordinates_from_surface(nb_triangles, nb_seed, triangles_mask=None,
+                                    triangles_weight=None, rand_gen=None):
+    """Generate random triangles_indices and trilinear_coord
+
+    Triangles_indices probability are weighted by triangles_weight,
+        for each triangles inside the given triangles_mask
+
+    Parameters
+    ----------
+    nb_triangles : int (n)
+        The amount of triangles in the mesh
+    nb_seed : int
+        The number of random indices and coordinates generated.
+    triangles_mask : [n] numpy array
+        Specifies which triangles should be chosen (or not)
+    triangles_weight : [n] numpy array
+        Specifies the weight/probability of choosing each triangle
+    random_seed : int
+        The seed for the random seed generator (numpy.random.seed).
+
+    Returns
+    -------
+    triangles_idx: [s] array
+        Randomly chosen triangles_indices
+    trilin_coord: [s,3] array
+        Randomly chosen trilinear_coordinates
+
+    See Also
+    --------
+    seeds_from_surface_coordinates, random_seeds_from_mask
+
+    Examples
+    --------
+    >>> mesh_polydata = load_polydata('mesh.vtk') # fury.io
```

Comment:
Doctests will not recognize these functions. you need to add the `import` above as usual.

```python
>>> from fury import io, utils
>>> mesh_polydata = io.load_polydata('mesh.vtk')
etc...
```